src/miner_restart_sup.erl

```diff
@@ -94,6 +94,19 @@ init([SigFun, ECDHFun]) ->
     JsonRpcPort = application:get_env(miner, jsonrpc_port, 4467),
     JsonRpcIp = application:get_env(miner, jsonrpc_ip, {127,0,0,1}),
 
+    DenylistWorker = case application:get_env(miner, denylist_key, undefined) of
```

Comment:
could you abstract this out of the supervisor by always adding the denylist worker to the supervision tree and then in the worker's init function return `ignore` if you don't have the right stuffs in the application config?

Reply:
Oh, I forgot about this, does it work for optional deps?

Reply:
like if we make the `xorf` app optional the way we've been talking about optional deps related to the tpm module branch? not sure but i've used this previously for enabling/disabling optional features in a supervision tree. if you're worried an optional dep could be missing could you make the init function do something like this:
```
case {application:get_env(miner,` denylist_config, undefined), code:ensure_loaded(xorf)} of
    {undefined, _} ->
        ignore;
    {Config, {module, xorf}} ->
        rest_of_the_worker_init,
        {ok, schedule_check(State, 0)
end
```

Reply:
it doesn't need to be an optional app, we just need to allow the denylist to not be configured

Reply:
Then it should work to make the worker accept an empty list in the supervisor childspec and then grab those config values in its init function or else return ignore if they’re not present. I did something similar previously where I checked if the child supervisor was `enabled` and if it was false I returned ignore and exited the process cleanly and in those cases the rest of that process’s config was omitted from the app env completely without any problems.